### DISCUSSION TITLE
Help me identify where a multi-NDI source low FPS issue is coming from

### DISCUSSION THREAD

**Discussion by Helgaiden** [DistroAV/DistroAV#644] [General]
hello all! I have been a big proponent of NDI but I have continuously run into an issue where when i try to stream our LAN parties and have multiple input sources from different computers (5-6 total), none of them seem to hit the OBS streaming computer smoothly. So here's the setup:

https://imgur.com/a/8LurCJr
That picture represents the living room of my house with 8 gamers. 4-5 of them have NDI output enabled at 1280x720 30fps in OBS. Sometimes there are 3-4 more computers off to the side (not pictured). 

Here us a room layout diagram:
https://imgur.com/a/3G7a3LV

now, here is the resulting stream: 
https://www.youtube.com/watch?v=G1jNLwmpchY

If you skip around in the stream, you'll see that it isn't exactly performing smoothly...except for that overhead cam shot. That overhead cam shot is my microsoft surface book's webcam outputting via NDI, over wireless AC wifi. So it isn't going thru either switch in the room, but rather wirelessly via wifi directly to the main router/switch i have in that side room. 

My computer was outputting my OBS over NDI the traditional way, but i also set a filter for the webcam, desktop audio, and microphone so they can have their own scene on the streaming PC seperate from my main OBS window. The result of this: the feed on the streaming PC from OBS NDI ran mcuh worse than everyone else who simply just enabled NDI then minimized their OBS and forgot about it. Every PC in the room is pretty high end, with mine being a 6-core/12-thread Xeon E5-1650v2 at 4.5ghz and most everyone else having a 6700k, 7700k, Ryzen 5, or 8700k. My Surface Book was set on an upper ledge my living room has, with the webcam pointing down at us for an overhead shot.

The streaming PC is USUALLY a system with a 4c/8t Xeon E5-1620 and 8gb ram. For this specific stream though, it was a Ryzen 2700 8-core with 16gb ram. Didn't seem to help all that much, i thought it would.

So what could i do to improve the smoothness of the NDI sources? Is my main gigabit switch not a good enough one? Too much bandwidth for it to properly handle? Its a Zyxel GS1100-16 : https://www.zyxel.com/us/en/products_services/8-10-16-24-port-GbE-Unmanaged-Switch-GS1100-Series/

Is my stream PC not being directly connected into the same switch as most the other computers part of the problem? The surface book'a webcam seems perfectly smooth (towards the end of the YouTube video stream linked above, especially, since people patching early on would screw things up) and it isn't going thru the main switch...
Currently the stream PC gets it's own port on the router/modem (Asus rt-68u I think), and the switch in the main LAN room gets it's own port (obviously) from one of then open ones on the Asus router/switch. Could this config be causing some sort of bottleneck?

any help would be greatly appreciated, thank you.

**Comment by CoDEmanX** [comment]
> A typical NDI stream consisting of 1080i HD video yields a data rate up to 100 Mbps per stream. 

https://support.newtek.com/hc/en-us/articles/217662708-NDI-Network-Bandwidth

Sounds like a gigabit network should have sufficient bandwidth for around 8-9 streams I guess? And if I understand this technology correctly, then the receiving computer has to decode these very lightly compressed streams, composite them (OBS canvas) and encode the resulting scene.

There are two possible bottlenecks: The decoding of 5+ streams, which might be over the top of what the CPU can handle with hardware acceleration, and the compositing of these rather high-resolution streams in OBS (not sure if this happens on the CPU or GPU). The encoding is done only once, but does of course tax your CPU a lot as well, assuming that you use software encoding with an encoding preset of rather high quality (x264 fast preset for instance).

In an up to date Windows 10, you should be able to see not only the CPU utilization in task manager, but also the GPU load for different types of tasks. Is something maxing out here?

**Comment by Helgaiden** [comment]
@CoDEmanX You know, i figured that the bottleneck was the hardware and really wanted to upgrade my CPU to a 10-core in my stream rig...then my buddy brought over his Ryzen 7 2700 8-core system to do this with, and i had the same issue. What was more confusing is some computers's NDI feeds ran better than others (mine ran the worse, but i also had main NDI output enabled and the dedicated NDI filter on my audio, microphone, and webcam to be used in a separate scene on the streaming PC). Then the Microsoft Surface Book that was on WiFi (bypassing the main gigabit switch) had a feed than ran perfectly smoothly. 

I started a thread over on reddit at the same time as this one hoping to get some help and there was a lot of info shared there too if you want to check it out.

https://www.reddit.com/r/Twitch/comments/9415ay/help_me_identify_where_a_multindi_source_low_fps/

**Comment by mipsou** [comment]
You not use switch managed and Multicast. Please respect this recommendation https://support.newtek.com/hc/en-us/articles/115001705074-NETWORK-SETTINGS.
You need a network expert to tune all switch

**Comment by Trouffman** [comment]
Closing - old thread.
Reason : Using an old plugin/OBS version or old-issue.

NB: If you end up here and have a similar issue, please open a troubleshooting discussion, with all your details, and only reference this discussion if it brings added-value to your issue.
Thank you
